cpp/src_prims/selection/knn.h

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cuda_utils.h"
+
+#include <faiss/gpu/GpuResources.h>
+#include <faiss/gpu/StandardGpuResources.h>
+#include <faiss/gpu/IndexProxy.h>
+#include <faiss/gpu/GpuIndexFlat.h>
+#include <faiss/gpu/GpuDistance.h>
+#include <faiss/Heap.h>
+
+#include <iostream>
+
+
+namespace MLCommon {
+namespace Selection {
+
+
+
+  /** Merge results from several shards into a single result set.
+   * @param all_distances  size nshard * n * k
```

Comment:
docstring doesn't correspond to function, or (more likely) needs to be updated for the changes of the refactor